src/nat.rs

```diff
@@ -32,8 +35,26 @@ pub fn get_public_ip_addr() -> Result<IpAddr, String> {
     }
 }
 
+pub fn udp_random_bind(start: u16, end: u16, tries: u32) -> Result<UdpSocket, String> {
+    let mut count = 0;
+    loop {
+        count += 1;
+
+        let rand_port = thread_rng().gen_range(start, end);
+        let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), rand_port);
+
+        match UdpSocket::bind(addr) {
+            Result::Ok(val) => break Result::Ok(val),
+            Result::Err(err) =>
+                if err.kind() != io::ErrorKind::AddrInUse || count >= tries {
+                    break Result::Err(err.to_string())
```

Comment:
Is this "Err(err.to_string())" correct way to propagate error to caller of udp_random_bind()?

Reply:
I get the following error. Maybe I am missing something.

50 |                     return Err(err);
   |                                ^^^
   |                                |
   |                                expected struct `std::string::String`, found struct `std::io::Error`
   |                                help: try using a conversion method: `err.to_string()`

Changing to return Err(err.to_string()) works. But is there any benefit of doing that above what's in the current patch?

Reply:
@pgarg66 your return type should be `io::Result<UdpSocket>` . Then you can passit io errors.  `io::Result` is a type alias for `Result<A, io::Error>`